### ISSUE TITLE
viya-pre-upgrade.yml hung at gathering facts when 2 Ansible versions installed and /etc/ansible/facts.d/* dir exists

### ISSUE THREAD

**Issue by afifdin** [sassoftware/viya-ark#49]
During pre-upgrade assessment, i'm encountered a situation where the ansible run would hung during gathering facts. At first it looks like SSH has broken-pipe but after further troubleshooting it shows that when ansible executed it looks for /etc/ansible/facts.d/* 

Seems like there is 2 ansible installed in the server, one using YUM and another one using PIP. Even after removed the ansible that installed using YUM also wouldn't help as it would not remove /etc/ansible/facts.d/* scripts

The workaround is to move the /etc/ansible/facts.d/* to other place and the assessment would completed as usual 

Further information can be found here: 
Track number: 7613042288: Deviation - Petronas - Ansible playbook hung when run viya-ark pre-upgrade scripts

If the above findings can be scripted into viya-ark that would be useful

Regards

**Comment by AlissaSchleich** [comment]
Thank you for your comment.   We are currently reproducing your issue and will assess a coding fix.

**Comment by kevinlinglesas** [comment]
@afifdin I'm having trouble attempting to reproduce this issue.   I have a Viya 3.4 depoloyment and wish to YUM install another version of ansible.    There is currently now /etc/ansible/facts.d directory present on the target (btw, single machine deployment).   Ansible 2.8.6 was used for the deployment.  When I try to install another version of ansible with `sudo yum install ansible' I get ansible 2.9.7 installed, but no /etc/ansible/facts.d directory is created.   I thought the implication was that the yum installed ansible created the facts.d directory.

In another Viya 3.5 deployment, I went through the steps (per deployment guide) to prepare for upgrade and could not reproduce the issue.   I did not attempt to install second ansible with yum on this deployment.  I did create /etc/ansible/facts.d dir, but it's existence didn't interfere with the playbook running.

Wondering if you had any more insight on how I might replicate your failing environment?
